src/components/ProfilePage.jsx

```diff
@@ -0,0 +1,152 @@
+import { useParams } from "react-router-dom";
+import { getRequest, putRequest } from "../utilites/apiRequests";
+import { useEffect, useState } from "react";
+import { ProfileIcon } from "./General/ProfileIcon";
+
+export const ProfilePage = () => {
+	const [user, setUser] = useState(null);
+	const [formData, setFormData] = useState(null);
+	const [successfullSubmit, setSuccesfullSubmit] = useState(false);
+	const { userId } = useParams();
+
+	const handleInput = (event) => {
+		const { name, value } = event.target;
+		setFormData({ ...formData, [name]: value });
+		setSuccesfullSubmit(false);
+	};
+
+	const handleSubmit = (event) => {
+		event.preventDefault();
+		putRequest(`/contact/${userId}`, formData).then((response) => {
+			setUser(response);
+			setSuccesfullSubmit(true);
+		});
+	};
+
+	useEffect(() => {
+		getRequest(`/contact/${userId}`).then((responseUser) => {
+			setUser(responseUser);
+			setFormData(responseUser);
+		});
+	}, [userId]);
+
+	return !user ? (
+		<p>Loading user...</p>
+	) : (
+		<main>
+			<div className="card">
+				<div className="border-bottom">
+					<div className="user-header">
+						<ProfileIcon user={user} />
+						<h1>{`${user.firstName} ${user.lastName}`}</h1>
+					</div>
+				</div>
+				<form className="profile-form">
```

Comment:
You have a pattern that repeats for each input field. Consider abstracting this into a reusable component or mapping over an array of field definitions. This will make your code DRYer and easier to maintain.

Example abstraction for form fields:

```javascript
const formFields = [
  { label: "First name", id: "firstName", name: "firstName" },
  { label: "Last name", id: "lastName", name: "lastName" },
  { label: "Email", id: "email", name: "email" },
  // Add other fields as necessary...
];

// In your form rendering:
{formFields.map((field) => (
  <li key={field.id}>
    <label htmlFor={field.id}>{field.label}*</label>
    <input
      type="text"
      id={field.id}
      name={field.name}
      onChange={handleInput}
      value={formData[field.name] || ""}
    />
  </li>
))}

```